### ISSUE TITLE
view does not resize anymore when soft keyboard opens

### ISSUE THREAD

**Issue by MaxjTechno** [Prototik/HoloEverywhere#431]
I'm porting my code using "pure" Android SDK widgets to holoeverywhere widgets and layout (1.5.0, master branch)
I've a problem with managing when softkeyboard opens inside my login view : my view does not resize anymore when one of the editext gets focus.

Before the porting to holoeverywhere everything was ok.
The view resized and the image + the 2 editexts went above the keyboard when it opened.

Now the keyboard covers them when it opens !
The view does not resize anymore.

In the AndroidManifest.xml the views' activity is declared with : android:windowSoftInputMode="stateUnchanged|adjustResize"

The view contains an image and two edittext, inside a LinearLayout (vertical) nested inside a Scrollview.
(There is also a ViewSwitcher wrapping everything : I have 2 versions of the same view).

I set the theme programmatically as I wrote here : 

http://stackoverflow.com/questions/16343979/holoeverywhere-how-to-programmatically-remove-at-runtime-the-action-bar-from-a

(the activity is the same)

I use :
setTheme(ThemeManager.DARK | ThemeManager.NO_ACTION_BAR);

I extend the org.holoeverywhere.app.Activity

I use the  xmlns:holo="http://schemas.android.com/apk/res-auto" annotation in the view's xml layout in the ViewSwitcher section declaration.

What do I have to do to have my login view resizing again when keyboard opens using Holoeverywhere ?

Thanks

Maxj

**Comment by Prototik** [comment]
LinearLayout inside ScrollView?
Stop, it's /should/ resized?
I make some test, tomorrow.

**Comment by MaxjTechno** [comment]
Yes.
The hierarchy is as below (see the xml snippet)
Inside the LinearLayout there is also a TableLayout.
You can omit the ViewSwitcher I think.
I'm re-writing again the view from zero to understand where the problem is, I'm getting crazy !!
thanks

ps
it's a login form, with username and password fields and submit button

```
  <ViewSwitcher
    xmlns:holo="http://schemas.android.com/apk/res-auto"
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:id="@+id/profileSwitcher"
    android:layout_width="fill_parent"
   android:layout_height="fill_parent">

      <ScrollView
        android:id="@+id/scrollview"
        android:layout_width="fill_parent"
        android:layout_height="fill_parent"
           android:isScrollContainer="true" >


            <LinearLayout
                android:layout_gravity="center"
                android:orientation="vertical"
                android:layout_width="fill_parent"
                android:layout_height="wrap_content"  >

                 <!-- now imageview and tablelayout are at the same level-->

                  <ImageView
                    android:layout_marginLeft="@dimen/firstPicLeft"
                    android:layout_marginRight="@dimen/firstPicRight"
                    android:layout_marginBottom="0.0dip"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_gravity="center"
                    android:src="@drawable/logo_transparent_01" />

                 <!-- tablelayout contains 3 rows : 2 EditText and 1 Button -->
                <TableLayout
                    android:id="@+id/tableLayout1"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_gravity="center"
                    android:layout_marginTop="@dimen/loginTableTop"
                    >

                       <TableRow
                            android:id="@+id/tableRow2"
                            android:layout_width="fill_parent"
                            android:layout_height="wrap_content" >


                                   <EditText
                                    android:minWidth="450.0dip"
                                    android:id="@+id/new_account_email_1"
                                    android:layout_marginTop="@dimen/loginTextTop"
                                    android:layout_marginBottom="@dimen/loginTextBottom"
                                    android:maxLines="1"
                                    android:scrollHorizontally="true"
                                    android:inputType="textEmailAddress"
                                    android:hint="@string/login_email_prompt"
                                    android:imeOptions="actionNext|flagNoExtractUi" />


                        </TableRow>

                                (and so on....)


                </TableLayout>          

            </LinearLayout>

    </ScrollView>
```

</ViewSwitcher>

**Comment by MaxjTechno** [comment]
Ok, I've found the line of code that does not allow my view to resize when softkeyboard opens.
It's this one :

```
    // Android 2.* incorrect process FULLSCREEN flag when we are modify
    // DecorView of Window. This hack using HoloEverywhere Slider
    if (VERSION.SDK_INT >= VERSION_CODES.HONEYCOMB) {
        ThemeManager.modify(ThemeManager.FULLSCREEN);
    }
```

I've adopted the skeleton of the Holoeverywhere Demo for initial setup of the framework, so I've introduced the  DemoApplication class, which I declare in my AndroidManifest.xml with

```
   android:name=".DemoApplication"
```

At runtime on my device (Android 4.1.1, 10" tablet) I have this values :

VERSION.SDK_INT =16
VERSION_CODES.HONEYCOMB =11

so I enter inside the if condition and the fullscreen is forced.
And the view does not resize when soft keyboard opens.

Is it correct to do that ?
I've commented it out now , and everything went ok.
Will I have some problems with Slider ? (I will reset it if needed....but if there is an issue, why isn't this line of code at the framework level ?)

But now my view resizes again

BUT

What I can notice now is that there are 2 new possible problems (in the same view, the login form),
( but I'll look after them later, since now I've to go to sleep :-)  )

I remember that I'm not modifying my code, I'm just introducing the framework, and I'm still at the first view, the login form, that contains 1 image (the logo), 2 EditText (username + password), 1 Button (submit)

Here they are :

1) when the apps start, the first EditText (username) automatically gets the focus (as before) and the softkeyboard opens (this was not happening before).
This when the app starts. I don't want it of course. I want the keyboard closed, as before.

In the declaration of the activity I use in fact :

```
   android:windowSoftInputMode="stateUnchanged|adjustResize"
```

so, what's the difference from before ?

2) the ImageView : when the soft keyboard is closed, I display a logo of 800x300, when the soft keyboard is open I display a smaller logo of 500x187 (I alternate two different views using the ViewSwitcher ).
Without holoeverywhere the switch is visually ok ("smooth" resize).
With holoeverywhere the switch produces weird flickering / flashing effects.
(especially in the switch from "keyboard open" to "keyboard closed", that is from the 500x187 image to the 800x300 image)
Without holoeverywhere it does not happen.

I do not want them of course.
It's the first screen of my app.

What's happening ?
The new superclass from which my activity now extends (org.holoeverywhere.app.Activity) is doing something that can cause this effect ?

But I go to sleep now....
If needed I'll open 2 new issues
(if you don't have already the answers for me.....)

Thanks again

Maxj

**Comment by MaxjTechno** [comment]
An update about the 2 "new" issues I talked about in my previous post.

1) softkeyboard opens automatically at launch :
SOLVED
My fault, I was looking at the wrong xml layout :-)

2) weird flickering / flashing effects :
NOT SOLVED, I bypassed it with a workaround

I think there is a problem using a viewswitcher with holoeverywhere (but I can be wrong of course)
I had to rewrite my login form to eliminate the use of the viewswitcher .

With the viewswitcher  switching 2 views containing an image, I had flickering / flashing effects, with the image moving quickly up and down at the moment of the switch.
Some strange "repainting" or "re-positioning" effects are being triggered at the moment of the switch.

I can produce a quick prototype if needed (can I upload it somewhere ?), I'll open a separate issue as soon as I'll be able to.
